src/main/java/org/broadinstitute/hellbender/utils/IndexRange.java

```diff
@@ -114,6 +115,66 @@ public void forEach(final IntConsumer lambda) {
         }
     }
 
+    /**
+     * Apply an int -> double function to this range, producing a double[]
+     *
+     * @param lambda the int -> double function
+     */
+    public double[] mapToDouble(final IntToDoubleFunction lambda) {
+        Utils.nonNull(lambda, "the lambda function cannot be null");
+        final double[] result = new double[size()];
+        for (int i = from; i < to; i++) {
+            result[i - from] = lambda.applyAsDouble(i);
+        }
+        return result;
+    }
+
+    /**
+     * Sums the values of an int -> double function applied to this range
+     *
+     * @param lambda the int -> double function
+     */
+    public double sum(final IntToDoubleFunction lambda) {
+        Utils.nonNull(lambda, "the lambda function cannot be null");
+        double result = 0;
+        for (int i = from; i < to; i++) {
+            result += lambda.applyAsDouble(i);
+        }
+        return result;
+    }
+
+    /**
+     * Apply an int -> int function to this range, producing an int[]
+     *
+     * @param lambda the int -> int function
+     */
+    public int[] mapToInteger(final IntUnaryOperator lambda) {
+        Utils.nonNull(lambda, "the lambda function cannot be null");
+        final int[] result = new int[size()];
+        for (int i = from; i < to; i++) {
+            result[i - from] = lambda.applyAsInt(i);
+        }
+        return result;
+    }
+
+    /**
+     * Find the elements of this range for which an int -> boolean predicate is true
+     *
+     * @param predicate the int -> boolean predicate
+     * @return
+     */
+    public int[] filter(final IntPredicate predicate) {
+        final int[] result = new int[size()];
```

Comment:
hmm, this seems like a potentially huge array allocation for what may turn out to be a small filtered result. Using an ArrayList (which grows as needed) would be better I think.